service/worker/failovermanager/workflow.go

```diff
@@ -473,15 +477,28 @@ func FailoverActivity(ctx context.Context, params *FailoverActivityParams) (*Fai
 			failedDomains = append(failedDomains, domain)
 			continue
 		}
-		updateRequest := &types.UpdateDomainRequest{
-			Name:              domain,
-			ActiveClusterName: common.StringPtr(params.TargetCluster),
-		}
-		if params.GracefulFailoverTimeoutInSeconds != nil {
-			updateRequest.FailoverTimeoutInSeconds = params.GracefulFailoverTimeoutInSeconds
+
+		var err error
+		// Use FailoverDomain API if reason is provided to properly track it
+		// Otherwise use UpdateDomain for backward compatibility and graceful failover support
```

Comment:
Let's not modify these worker failovers just yet; they're used for something a bit more complex with the failover worker, it's ok for the reason field not to be included here.